src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.h

```diff
@@ -16,115 +16,249 @@
+    GO_EADRIC_S_CACHE                               = 195374,
+    GO_EADRIC_S_CACHE_H                             = 195375,
 
-    GO_PALETRESS_LOOT           = 195323,
-    GO_PALETRESS_LOOT_H            = 195324
+    GO_CONFESSOR_S_CACHE                            = 195323,
+    GO_CONFESSOR_S_CACHE_H                          = 195324
 };
 
 enum Vehicles
 {
-    //Grand Champions Alliance Vehicles
+    // Grand Champions Alliance Vehicles
     VEHICLE_MARSHAL_JACOB_ALERIUS_MOUNT             = 35637,
     VEHICLE_AMBROSE_BOLTSPARK_MOUNT                 = 35633,
     VEHICLE_COLOSOS_MOUNT                           = 35768,
     VEHICLE_EVENSONG_MOUNT                          = 34658,
     VEHICLE_LANA_STOUTHAMMER_MOUNT                  = 35636,
-    //Faction Champions (ALLIANCE)
-    VEHICLE_DARNASSIA_NIGHTSABER                    = 33319,
-    VEHICLE_EXODAR_ELEKK                            = 33318,
-    VEHICLE_STORMWIND_STEED                         = 33217,
-    VEHICLE_GNOMEREGAN_MECHANOSTRIDER               = 33317,
-    VEHICLE_IRONFORGE_RAM                           = 33316,
-    //Grand Champions Horde Vehicles
+    // Faction Champions (Alliance)
+    VEHICLE_DARNASSUS_CHAMPION                      = 35332,
+    VEHICLE_EXODAR_CHAMPION                         = 35330,
+    VEHICLE_STORMWIND_CHAMPION                      = 35328,
+    VEHICLE_GNOMEREGAN_CHAMPION                     = 35331,
+    VEHICLE_IRONFORGE_CHAMPION                      = 35329,
+    // Grand Champions Horde Vehicles
     VEHICLE_MOKRA_SKILLCRUSHER_MOUNT                = 35638,
     VEHICLE_ERESSEA_DAWNSINGER_MOUNT                = 35635,
     VEHICLE_RUNOK_WILDMANE_MOUNT                    = 35640,
     VEHICLE_ZUL_TORE_MOUNT                          = 35641,
     VEHICLE_DEATHSTALKER_VESCERI_MOUNT              = 35634,
-    //Faction Champions (HORDE)
-    VEHICLE_FORSAKE_WARHORSE                        = 33324,
-    VEHICLE_THUNDER_BLUFF_KODO                      = 33322,
-    VEHICLE_ORGRIMMAR_WOLF                          = 33320,
-    VEHICLE_SILVERMOON_HAWKSTRIDER                  = 33323,
-    VEHICLE_DARKSPEAR_RAPTOR                        = 33321,
-
-    VEHICLE_ARGENT_WARHORSE                         = 35644,
+    // Faction Champions (Horde)
+    VEHICLE_UNDERCITY_CHAMPION                      = 35327,
+    VEHICLE_THUNDER_BLUFF_CHAMPION                  = 35325,
+    VEHICLE_ORGRIMMAR_CHAMPION                      = 35314,
+    VEHICLE_SILVERMOON_CHAMPION                     = 35326,
+    VEHICLE_SENJIN_CHAMPION                         = 35323,
+    // Player vehicles
+    VEHICLE_ARGENT_WARHORSE_COSMETIC                = 35644,
+    VEHICLE_ARGENT_WARHORSE                         = 36557,
     VEHICLE_ARGENT_BATTLEWORG                       = 36558,
-
+    VEHICLE_ARGENT_BATTLEWORG_COSMETIC              = 36559,
+    // The Black Knight's vehicle
     VEHICLE_BLACK_KNIGHT                            = 35491
 };
 
-#endif
+enum WaypointData
```

Comment:
That commit make errors for me :

\src\server\scripts\northrend\crusaderscoliseum\trialofthechampion\trial_of_the_champion.h(224): error C2011: 'WaypointData': 'struct' type redefinition (compiling source file \src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_argent_challenge.cpp)
\src\server\game\Movement\Waypoints\WaypointManager.h(35): note: see declaration of 'WaypointData' (compiling source file \src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_argent_challenge.cpp)
\src\server\scripts\northrend\crusaderscoliseum\trialofthechampion\trial_of_the_champion.h(224): error C2011: 'WaypointData': 'struct' type redefinition (compiling source file \src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_black_knight.cpp)
\src\server\game\Movement\Waypoints\WaypointManager.h(35): note: see declaration of 'WaypointData' (compiling source file \src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_black_knight.cpp)
\src\server\scripts\northrend\crusaderscoliseum\trialofthechampion\trial_of_the_champion.h(224): error C2011: 'WaypointData': 'struct' type redefinition (compiling source file src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\instance_trial_of_the_champion.cpp)
\src\server\game\Movement\Waypoints\WaypointManager.h(35): note: see declaration of 'WaypointData' (compiling source file \src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\instance_trial_of_the_champion.cpp)
\src\server\scripts\northrend\crusaderscoliseum\trialofthechampion\trial_of_the_champion.h(224): error C2011: 'WaypointData': 'struct' type redefinition (compiling source file \src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp)
\src\server\game\Movement\Waypoints\WaypointManager.h(35): note: see declaration of 'WaypointData' (compiling source file \src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp)
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(760): error C2065: 'DATA_GENERATE_WAYPOINTS_FOR_BOSS_1': undeclared identifier
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(760): error C2131: expression did not evaluate to a constant
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(760): note: failure was caused by non-constant arguments or reference to a non-constant symbol
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(760): note: see usage of 'DATA_GENERATE_WAYPOINTS_FOR_BOSS_1'
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(777): error C2065: 'DATA_GENERATE_WAYPOINTS_FOR_BOSS_2': undeclared identifier
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(777): error C2131: expression did not evaluate to a constant
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(777): note: failure was caused by non-constant arguments or reference to a non-constant symbol
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(777): note: see usage of 'DATA_GENERATE_WAYPOINTS_FOR_BOSS_2'
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(792): error C2065: 'DATA_GENERATE_WAYPOINTS_FOR_BOSS_3': undeclared identifier
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(792): error C2131: expression did not evaluate to a constant
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(792): note: failure was caused by non-constant arguments or reference to a non-constant symbol
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(792): note: see usage of 'DATA_GENERATE_WAYPOINTS_FOR_BOSS_3'
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(807): error C2065: 'DATA_GENERATE_WAYPOINTS_FOR_ADDS': undeclared identifier
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(807): error C2131: expression did not evaluate to a constant
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(807): note: failure was caused by non-constant arguments or reference to a non-constant symbol
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(807): note: see usage of 'DATA_GENERATE_WAYPOINTS_FOR_ADDS'
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(814): error C2065: 'DATA_GENERATE_WAYPOINTS_FOR_BOSS_3': undeclared identifier
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(760): error C2051: case expression not constant
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(777): error C2051: case expression not constant
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(792): error C2051: case expression not constant
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(807): error C2051: case expression not constant
\src\server\scripts\Northrend\CrusadersColiseum\TrialOfTheChampion\boss_grand_champions.cpp(812): warning C4060: switch statement contains no 'case' or 'default' label